### ISSUE TITLE
Remove usage of `np.unicode_`?

### ISSUE THREAD

**Issue by nateybear** [jeffgortmaker/pyblp#168]
Hi! I'm working with a formulation that includes a string variable as a fixed effect to be absorbed. Trying to work with this object gives me an error: 
```
PatsyError('Error evaluating factor: AttributeError: `np.unicode_` was removed in the NumPy 2.0 release. Use `np.str_` instead.')
```
It looks like there's only one usage of `np.unicode_` in the codebase (and none in Patsy), so I'm guessing this should be bumped to `np.str_` conditional on a numpy version check.

I would submit the change myself as a pull request but I'm calling pyblp through Julia and PyCall doesn't do a great job reporting the stack trace, hence I'm not 100% sure that this is where my error is coming from. If you agree this is something that should change, however, I'm happy to contribute a pull request.

Thanks!
Nate

ETA: If I recode as an integer, i.e. `df.col = indexin(df.col, unique(df.col))`, I can avoid this issue, so I would mark as low priority.

**Comment by jeffgortmaker** [comment]
Thanks for the clear bug report and finding the source of the issue! I'll just push a quick fix myself.

It's a busy time, so might be a little while until it makes it into a stable version. But until then you can use the dev code in this repo, or recode as an integer.

I'm curious if you're finding PyCall otherwise okay with using PyBLP? I've heard good things about using it through R, but less about from Julia.

**Comment by nateybear** [comment]
Thanks a lot! I can manage with coercing to ints for now so I'm happy to wait on the version bump.

I find PyCall quite enjoyable overall, and pyblp works with it. PyCall automatically marshals primitive types and collections (arrays being coerced to numpy series), you can explicitly marshal dataframes using the `Pandas.jl` package, and you can work directly with Python modules/classes in Julia without marshaling. The logit tutorial looks like

```julia
using PyCall, CSV, DataFrames
import Pandas

pyblp = pyimport("pyblp")

product_data = CSV.read(pyblp.data.NEVO_PRODUCTS_LOCATION, DataFrame)

logit_formulation = pyblp.Formulation("prices", absorb = "C(product_ids)") # well, coerce product_ids to int for now :)
problem = pyblp.Problem(logit_formulation, Pandas.DataFrame(product_data))

problem.solve()
```

The two pain points are dealing with exceptions (limited stack trace info, as I alluded to above) and managing the python installation—PyCall ships with a miniconda installation that you can manage or you can point to the system python, which is not too hard in general but a little confusing for non-technical folks. `Pandas.jl` also doesn't allow converting missing data to pandas dataframes, which is an understandable design choice but a little annoying.

Anyways, thanks for the responsiveness and the great package design. I hear whispers that pyblp on the GPU is coming, so I'm excited to see how that implementation shakes out!

**Comment by jeffgortmaker** [comment]
Thanks for the info about using PyCall -- that's useful for when I talk to Julia users.
